packages/beacon-node/src/chain/initState.ts

```diff
@@ -40,6 +41,22 @@ export async function persistAnchorState(
 ): Promise<void> {
   if (anchorState.slot === GENESIS_SLOT) {
     const genesisBlock = createGenesisBlock(config, anchorState);
+    const blockRoot = config.getForkTypes(GENESIS_SLOT).BeaconBlock.hashTreeRoot(genesisBlock.message);
+
+    const latestBlockHeader = ssz.phase0.BeaconBlockHeader.clone(anchorState.latestBlockHeader);
+
+    if (ssz.Root.equals(latestBlockHeader.stateRoot, ZERO_HASH)) {
+      latestBlockHeader.stateRoot = anchorState.hashTreeRoot();
+    }
+
+    const latestBlockRoot = ssz.phase0.BeaconBlockHeader.hashTreeRoot(latestBlockHeader);
```

Comment:
is there a case where this is not a `ZERO_HASH`? we might just wanna do

```suggestion
    const blockRoot = config.getForkTypes(GENESIS_SLOT).BeaconBlock.hashTreeRoot(genesisBlock.message);
    const latestBlockRoot = ssz.phase0.BeaconBlockHeader.hashTreeRoot({
      ...anchorState.latestBlockHeader.toValue(),
      stateRoot: anchorState.hashTreeRoot(),
    });
```

Reply:
> is there a case where this is not a `ZERO_HASH`? we might just wanna do

Hmmm, not sure. But here we download a state from external source https://github.com/ChainSafe/lodestar/blob/99033e3bd9e7a9e73de99f2a6b3e299f59c76440/packages/cli/src/cmds/beacon/initBeaconState.ts#L192 which can potentially have some non-zero byte state root if some issue occurred on their side.

Reply:
but this code path is only executed if it's a genesis state

https://github.com/ChainSafe/lodestar/blob/99033e3bd9e7a9e73de99f2a6b3e299f59c76440/packages/beacon-node/src/chain/initState.ts#L41

but can keep the check either way, doesn't hurt I guess